aibolit/patterns/supermethod/supermethod.py

```diff
@@ -20,27 +35,16 @@ def value(self, filename: str):
         """
         results = []
         source_code = read_text_with_autodetected_encoding(filename)
-        tree = parse(source_code)
         lines = source_code.splitlines()
-        for _, method_decl_node in tree.filter(MethodDeclaration):
-            code_line = method_decl_node.position.line
-            for _, super_method_inv in method_decl_node.filter(SuperMethodInvocation):
+        tree = JavaPackage(filename)
+        for method_decl_node in tree.nodes_by_type(ASTNodeType.METHOD_DECLARATION):
```

Comment:
It sounds to me, that outer loop is redundant and just brings repetition of code lines, which was tried to removed by using set. If you iterate over all super method invocations, isn't it enough to find all its lines? By the way, there are probably will be no source code available for patterns in the future. We shouldn't stick to using it.

Reply:
defenitely agree. Fixed it in the last commit